libvast/vast/system/disk_monitor.hpp

```diff
@@ -0,0 +1,46 @@
+// Experimental actor to monitor disk space usage
+#pragma once
+
+#include "vast/fwd.hpp"
+#include "vast/path.hpp"
+#include "vast/system/archive.hpp"
+
+#include <caf/typed_actor.hpp>
+
+namespace vast::system {
+
+// clang-format off
+using disk_monitor_type = caf::typed_actor<
+  caf::reacts_to<atom::ping>,
+  caf::reacts_to<atom::erase>
+>;
+// clang-format on
+
+struct disk_monitor_state {
+  /// The path to the database directory.
+  vast::path dbdir;
+
+  /// When current disk space is above the high water mark, stuff
+  /// is deleted until we get below the low water mark.
+  size_t high_water_mark;
+  size_t low_water_mark;
+
+  /// Whether an erasing run is currently in progress.
+  bool purging;
+
+  /// Node handle of the ARCHIVE.
+  archive_type archive;
+
+  /// Node handle of the INDEX.
+  caf::actor index;
+
+  constexpr static const char* name = "disk_monitor";
+};
+
+disk_monitor_type::behavior_type
```

Comment:
A comment here would be nice, just about the rough functionality this actor provides.